BuildMacros.cmake

```diff
@@ -114,15 +114,15 @@ macro(setup_library)
     #NOT SURE IF NEEDED
     #   do I need to remove the source files that were compiled into the 
     #   event bus dictionary? Or is it okay for them to be compiled twice?
-    set(link_against_dictionary "NO")
-    foreach(src ${SRC_FILES})
-        list(FIND event_sources ${src} index)
-        if(${index} GREATER 0)
```

Comment:
@omar-moreno I only found that this double-compiling was causing errors when I tried to run a basic sim script and link the simulation and event libraries with the framework. If you are working outside the container, also be aware that Ubuntu is more restrictive, so you might get away with something that the container won't let you get away with.